Echo support

malberts opened the pull request:
Fixes: #3
Fixes: #83
Fixes: #90

s7eph4n commented:
When introducing new components, remember to also update https://github.com/ProfessionalWiki/chameleon/blob/master/layouts/layout.rng
It is admittedly not straight-forward, but it allows to check that a layout file is at least formally correct.

malberts commented:
Thanks for reminding me. I saw that file and was wondering about how to update it. Is there a specific tool you used before, or do you edit it by hand?

s7eph4n commented:
I always edited it by hand. I don't know if there is an editor available.

malberts commented:
@s7eph4n I updated layouts.rng. Does that look about right?

s7eph4n commented:
I think it looks good. 👍

malberts commented:
@JeroenDeDauw I rewrote a bunch of the code here. Primarily to simplify it and hopefully also to improve the code a bit.

Changes:
* Removed the standalone EchoIcons component. Nothing else gets this special treatment and it's more code to maintain. If somebody ever asks for this I'll look at it again
* Consistent defaults and options for standalone and Navbar PersonalTools
  * Standalone PersonalTools can show icons or links
  * Navbar PersonalTools can show icons _next to_ the dropdown, or links _in_ the dropdown
* Default behavior is to show the icons, since that is how Echo works by default (the links mode was a hack due to the icon support being broken previously)

I'm not going to make big changes again. If the code is fine I'm going to start the 3.2.0 release after merging.

JeroenDeDauw commented:
Needs a rebase

malberts commented:
Rebased. Do you want me to squash the commits further?
